### ISSUE TITLE
Connection to WDA timed out when i try to turn off the wlan on ipad

### ISSUE THREAD

**Issue by boan85123** [appium/python-client#424]
## The problem

The WDA can't be connected in the following scenario:
1. login my app, and click the inner link to go to network setting page of my pad;
2. turn on airplane mode and click breadcrumb to go back my app;
3. go on doing some test in my app.

When i switch between my app and network setting page, the WDA will time out.

## Environment
- Appium version (or git revision) that exhibits the issue: Appium 1.14.0-beta.1
- Last Appium version that did not exhibit the issue (if applicable): None
- Desktop OS/version used to run Appium: MacOS 10.14.3
- Node.js version (unless using Appium.app|exe): v11.10.0
- Mobile platform/version under test: IOS 12.3.1
- Real device or emulator/simulator: Both real device and emulator.
- Appium CLI or Appium.app|exe: 

## Link to Appium Logs

[logs](https://gist.githubusercontent.com/boan85123/0d162e7be76b7c202df7c0d246e85661/raw/6b1d757a0313e104de880b33f617997230ed5345/gistfile1.txt)
[device_log](https://gist.githubusercontent.com/boan85123/0d162e7be76b7c202df7c0d246e85661/raw/3adff0597a1ef1a4463f38a9da4fd423fe82efc9/gistfile2.txt)

**Comment by mykola-mokhnach** [comment]
Such behaviour is expected. The only possible workaround is to restart the driver after the network switch

**Comment by boan85123** [comment]
@mykola-mokhnach could you tell me how to restart the driver in code?

**Comment by mykola-mokhnach** [comment]
you simply quit it and then create a new session

**Comment by boan85123** [comment]
@mykola-mokhnach, i try to switch between my app and network setting. The WDA will time out too. I haven't changed my network status at all. Do you know why the WDA quits too even though i haven't changed my network status?

**Comment by boan85123** [comment]
@mykola-mokhnach, i use appium with ruby, may i create a new session with following code:
```ruby
desired_caps = {
  caps: {
    platformName:  'iOS',
    platformVersion: '10.2',
    deviceName:    'iPhone 6',
    app:           APP_PATH,
    automationName: "XCUITest"
  }
}

Appium::Driver.new(desired_caps).start_driver```  
Is it right? I'm a novice on appium, could you please give me some comments？

**Comment by KazuCocoa** [comment]
Your code is right.
I haven't met your case https://github.com/appium/python-client/issues/424#issuecomment-517176163 , but what about removing WDA once from your device completely once?
Rebooting your device once also sometimes helped.
(Vanilla XCTest also sometimes need to reboot devices if similar strange behaviour happenes..)

---

btw, I was able to conduct below command with the latest master. They have no much differences between your WDA and mine. WDA did not hang.

```
[1] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> @@driver.activate_app 'com.apple.Preferences'
=> {}
[2] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> e = @@driver.find_element :class, 'XCUIElementTypeSwitch'
=> #<Selenium::WebDriver::Element:0x..f0fc1d0956fa5e9a id="5B010000-0000-0000-EC0F-000000000000">
[3] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> e.click # turn airplain mode on
=> ""
[4] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> e.click # turn airplain mode off
=> ""
[5] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> @@driver.activate_app 'com.apple.Preferences'
=> {}
[6] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> e = @@driver.find_element :class, 'XCUIElementTypeSwitch'
=> #<Selenium::WebDriver::Element:0x..f0fc1d0956fa5e9a id="5B010000-0000-0000-EC0F-000000000000">
[7] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> e.click
=> ""
[8] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> @@driver.activate_app 'com.apple.measure'
=> {}
[9] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> @@driver.activate_app 'com.apple.Preferences'
=> {}
[10] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> e = @@driver.find_element :class, 'XCUIElementTypeSwitch'
=> #<Selenium::WebDriver::Element:0x..f0fc1d0956fa5e9a id="5B010000-0000-0000-EC0F-000000000000">
[11] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> e.click
=> ""
[12] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> @@driver.activate_app 'com.apple.Preferences'
=> {}
[13] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> @@driver.activate_app 'com.apple.measure'
=> {}
[14] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> @@driver.activate_app 'com.apple.measure'
=> {}
[15] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> @@driver.activate_app 'com.apple.Preferences'
=> {}
[16] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> @@driver.activate_app 'com.apple.Preferences'
=> {}
[17] pry(#<AppiumLibCoreTest::Ios::DeviceTest>)> @@driver.activate_app 'com.apple.videos'
```

**Comment by boan85123** [comment]
@KazuCocoa, thanks for your comment. I've tried it again after reboot my ipad, and the WDA was time out too. I think the different with us is the method "activate_app". I use the inner link of my app to go to the network config page instead of "activate_app". Let me try again with "activate_app".

**Comment by KazuCocoa** [comment]
Ah, thanks.
I think it is no matter via the inner link or `activate_app` (via XCTest feature).

Closes once since this is not functional issue.
Feel free to create a new issue in https://github.com/appium/appium (Appium server related) or ruby client repo for ruby related issue.

**Comment by boan85123** [comment]
@KazuCocoa , yes, the method "activate_app" still doesn't work for me. I'll try to start a new session in my code. Thanks a lot.

**Comment by KazuCocoa** [comment]
You can try the current beta, v1.15.0-beta.0, which has communication improvements between Appium and iOS devices. Not sure your main issue is related to such low layer one, but I think it's worth to try for you.
